File: sdk/unity/sdk/Assets/KingstonDynamics/DragonSong/Protocol/Messaging/Type/Authentication/AuthenticationReconnect.cs
using KingstonDynamics.DragonSong.Protocol.Messaging.Data;
using KingstonDynamics.DragonSong.Protocol.Utils;

namespace KingstonDynamics.DragonSong.Protocol.Messaging.Type.Authentication
{
    public class AuthenticationReconnect : Message
    {
        private static readonly MessageType Type = MessageType.AUTHENTICATION_LOGIN;

        public AuthenticationReconnect(string auditId) : base((int)Type, auditId)
        {
            // Empty
        }

        public AuthenticationReconnect() : this(Keyinator.GenerateGUID())
        {
            // Empty
        }

        public AuthenticationReconnect(Readinator reader) : base(reader)
        {
            // Empty
        }
    }
}

File: src/protocol/src/main/java/com/kingstondynamics/dragonsong/protocol/message/type/authentication/AuthenticationReconnect.java
package com.kingstondynamics.dragonsong.protocol.message.type.authentication;

import com.kingstondynamics.dragonsong.protocol.message.Message;
import com.kingstondynamics.dragonsong.protocol.message.data.MessageType;
import com.kingstondynamics.dragonsong.protocol.utils.Keyinator;

public class AuthenticationReconnect extends Message {
    
    private final static MessageType type = MessageType.AUTHENTICATION_RECONNECT;
    
    public AuthenticationReconnect(String auditId) {
        super(type.getType(), auditId);
    }
    
    public AuthenticationReconnect() {
        this(Keyinator.generateGUID());
    }
}
